internal/requestutil/util.go

```diff
@@ -0,0 +1,51 @@
+package requestutil
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func parseIP(ipStr string) net.IP {
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		log.Warnf("invalid remote IP address: %q", ipStr)
```

Comment:
question: I've sometimes wondered why we aren't using the `dcontext.Logger` but instead use the "raw" `logrus`

Reply:
I went to go plumb contexts through (from `(*http.Request).Context()`) but realized that requestutil importing dcontext would be an import cycle as dcontext already imports requestutil. Frankly the requestutil functions are not amazing (parse the IP address only to throw away the parsed form and return the source string representation?!) so I think that the the import cycle would best be broken by redesigning the requestutil functions such that the caller can do the logging, if it chooses to.